apps/labs/posts/cupy-czi-grant-year1.md

```diff
@@ -0,0 +1,121 @@
+  <figcaption style={{textAlign: "center"}}>The four random input signals from the previous figure with added Gaussian noise.</figcaption>
+</figure>
+
+Now that we’ve generated our signals, time for the fun part! Let’s do some filtering! In doing so, we’ll extract the noise from each signal using the new signal parallel-processing APIs available in CuPy, namely `lfilter` and `filtfilt`. For purposes of comparison, we’ll compare applying `lfilter` once, applying `lfilter` twice, and filtering in both forward and reverse directions using `filtfilt`.
+
+
+```python
+# Create a third-order, Butterworth low-pass filter with cutoff at 0.05.
+b, a = signal.butter(3, 0.05)
+b = b.astype(cp.float32)
+a = a.astype(cp.float32)
+
+# Compute the steady-state response of the filter.
+zi = signal.lfilter_zi(b, a)
+zi = zi.reshape(1, -1)
+zi = zi.astype(cp.float32)
+
+# Filter out the 1.25Hz and 3.85Hz components and remove the noise on all signals in parallel.
+z, _ = signal.lfilter(b, a, xn, zi=zi * xn[:, 0, None])
+
+# Apply the filter again.
+z2, _ = signal.lfilter(b, a, z, zi=zi*z[:, 0, None])
+
+# Use filtfilt on all signals in parallel.
+y = signal.filtfilt(b, a, xn)
+```
+
+A plot comparing filtering results is displayed below. By visual inspection (and rather cursory analysis!), we can conclude that `filtfilt` does a better job at recovering the original signals.
+
+<figure>
+  <img src="/posts/cupy-czi-grant-year1/filtered_signals.png" alt="A grid of four plots, arranged in a 2x2 configuration. Each plot shows the result of taking a noisy signal, applying the lfilter and filtfilt CuPy in the following order: lfilter once, lfilter twice and filfilt. The filtfilt results reconstruct the original signal with higher fidelity." />
+  <figcaption style={{textAlign: "center"}}>Filtering results using the new lfilter and filtfilt CuPy APIs. Each result is compared against the original, non-noisy signal.</figcaption>
+</figure>
+
+As part of this effort, we made several other improvements to other CuPy modules, such as adding special functions for elliptic filters, implementing the matrix exponential in the `cupyx.scipy.linalg` namespace, and implementing KD-Trees in `cupyx.scipy.spatial`.
+
+This work would not be possible without the help of the [cuSignal](https://github.com/rapidsai/cusignal) team at the NVIDIA RAPIDS initiative, where more than 36 APIs will be migrated from their project into CuPy, thus unifying the GPU signal processing efforts currently available in the ecosystem. As part of this migration, cuSignal will be deprecated and replaced by the new `cupyx.scipy.signal` module calls. The calls will be interchangeable by simply replacing the `cusignal` imports with imports from `cupyx.scipy.signal`. For more information about this collaboration, we invite you to visit the [RAPIDS blog](https://medium.com/rapids-ai).
```

Comment:
Can you link to the actual blog post from the RAPIDS team?

Reply:
I think they haven't published it yet, that's why I left the link to their overall blog instead to the particular page, since it does not exist yet. Let me confirm with @awthomp if the blog post is now available

Reply:
It isn't there yet. @awthomp do you know if is it expected to arrive shortly? If so, we can wait for it - or we can just leave out this sentence, and add it back in if and when that post appears. Right now the reader is invited to go look for something that isn't there, which would be good to avoid.

Reply:
I'll remove the sentence for the sake of publishing